testing/vault/vault.initd

```diff
@@ -2,17 +2,28 @@
 
 VAULT_LOG_FILE="/var/log/${SVCNAME}.log"
 
-command=/usr/sbin/${SVCNAME}
-command_args="${vault_opts}"
-command_background="true"
-start_stop_daemon_args="--user ${SVCNAME}:${SVCNAME}  --stdout $VAULT_LOG_FILE --stderr $VAULT_LOG_FILE"
-pidfile="/run/${SVCNAME}.pid"
+name=vault
+description="Vault is a tool for securely accessing secrets"
+daemon=/usr/sbin/$name
+daemon_user=$name
+daemon_group=$name
 
 start_pre() {
 	checkpath -f -m 0644 -o ${SVCNAME}:${SVCNAME} "$VAULT_LOG_FILE"
 }
 
 depend() {
-        need net
-        after firewall
+	need net
+	after firewall
 }
+
+start() {
+	ebegin "Starting ${SVCNAME}"
+	start-stop-daemon --start --quiet \
+		rm --pidfile /var/run/${name}.pid \
+		--user ${daemon_user} --group ${daemon_group} \
+		-b --stdout $VAULT_LOG_FILE --stderr $VAULT_LOG_FILE \
+		-k 027 --exec ${daemon} -- ${vault_opts}
+	eend $?
+}
```

Comment:
I had issues with service start in my testing. It would assume that the service started and then report that it crashed (instead of start failed).

In my setup I setup "consul" service as backend. So if "consul" is not running vault should fail to start.